crutest/src/main.rs

```diff
@@ -2132,18 +2130,23 @@ async fn perf_workload(
     for _ in 0..read_loop {
         let big_start = Instant::now();
         for _ in 0..count {
+            let read_buffers: Vec<Buffer> =
```

Comment:
This will have a measurement impact, since we're now doing allocation in the inner loop.

Maybe allocate `read_buffers` once outside (like before), then re-collect them when joining futures?

e.g. around L2145

```rust
let mut done_buffers = Vec::with_capacity(io_depth);
for future in read_futures {
    done_buffers.push(future.await?);
}
read_buffers = done_buffers;
```

Reply:
Done in 7819007, nice catch :)